src/builtins.c

```diff
@@ -778,6 +778,32 @@ JL_CALLABLE(jl_f__call_in_world)
     return ret;
 }
 
+JL_CALLABLE(jl_f__call_in_world_nonpure)
```

Comment:
We call the other copy of this code `jl_f__apply_pure`, so seems confusing that we would call this one `nonpure`

Reply:
Well, this does not set `in_pure_callback` unlike the `_apply_pure` and in fact unsets it if it is set. But yes, there could be a better name. Any suggestions?